app/gui2/vite.config.ts

```diff
@@ -63,7 +59,7 @@ export default defineConfig({
     ...getDefines(localServerPort),
     IS_CLOUD_BUILD: JSON.stringify(IS_CLOUD_BUILD),
     PROJECT_MANAGER_URL: JSON.stringify(projectManagerUrl),
-    YDOC_SERVER_URL: IS_POLYGLOT_YDOC_SERVER ? JSON.stringify('defined') : undefined,
+    YDOC_SERVER_URL: JSON.stringify(POLYGLOT_YDOC_SERVER),
```

Comment:
I guess this change is one of the causes for such a weird config check. The `undefined` value should not be stringified here.
```suggestion
    YDOC_SERVER_URL: POLYGLOT_YDOC_SERVER ? JSON.stringify(POLYGLOT_YDOC_SERVER) : undefined,
```

Reply:
`JSON.stringify(undefined) === undefined` anyway. I was using this special `'undefined'` string because the config only allows having string values. I refactored the logic